src/components/Edges/wrapEdge.tsx

```diff
@@ -134,6 +134,20 @@ export default (EdgeComponent: ComponentType<EdgeProps>) => {
     const onEdgeUpdaterMouseEnter = useCallback(() => setUpdating(true), [setUpdating]);
     const onEdgeUpdaterMouseOut = useCallback(() => setUpdating(false), [setUpdating]);
 
+    const radiusEdgeUpdater = 10;
+
+    const shiftX = (position: string, x: number) => {
+      if (position === 'right') return x + radiusEdgeUpdater;
+      if (position === 'left') return x - radiusEdgeUpdater;
+      return x;
+    }
+
+    const shiftY = (position: string, y: number) =>{
+      if (position === 'bottom') return y + radiusEdgeUpdater;
+      if (position === 'top') return y - radiusEdgeUpdater;
+      return y;
+    }
+
```

Comment:
Couldn't we define `radiusEdgeUpdater`, `shiftX` and `shiftY` outside of the component. This would prevent the reinitialization of the functions on every render.

Reply:
yep)) let me introduce `EdgeAnchor` component
```jsx
import React from 'react';

import { WrapEdgeProps, Position } from '../../types';

const radiusEdgeUpdater = 10;



const shiftX = (position: Position, x: number) => {
  if (position === Position.Right) return x + radiusEdgeUpdater;
  if (position === Position.Left) return x - radiusEdgeUpdater;
  return x;
}

const shiftY = (position: Position, y: number) =>{
  if (position === Position.Bottom) return y + radiusEdgeUpdater;
  if (position === Position.Top) return y - radiusEdgeUpdater;
  return y;
}

export interface EdgeAnchorProps extends WrapEdgeProps {
  isSource?: boolean
}

export const EdgeAnchor = ({
  sourceX,
  sourceY,
  targetX,
  targetY,
  sourcePosition,
  targetPosition,
  isSource = true
}: EdgeAnchorProps): JSX.Element | null => {
  return (
    <circle
      className="react-flow__edgeupdater"
      cx={isSource ? shiftX(sourcePosition, sourceX) : shiftX(targetPosition, targetX)}
      cy={isSource ? shiftY(sourcePosition, sourceY) : shiftY(targetPosition, targetY)}
      r={radiusEdgeUpdater}
      stroke="transparent"
      fill="transparent"
    />
  );
}
```

Reply:
or maybe use 2 components `EdgeAnhcorStart` & `EdgeAnhcorEnd` with only `WrapEdgeProps` without `isSource` ?
@moklick  what do you think?

Reply:
```jsx
import React from 'react';

import { WrapEdgeProps, Position } from '../../types';

const radiusEdgeUpdater = 10;

const shiftX = (position: Position, x: number) => {
  if (position === Position.Right) return x + radiusEdgeUpdater;
  if (position === Position.Left) return x - radiusEdgeUpdater;
  return x;
}

const shiftY = (position: Position, y: number) =>{
  if (position === Position.Bottom) return y + radiusEdgeUpdater;
  if (position === Position.Top) return y - radiusEdgeUpdater;
  return y;
}

export const EdgeAnchorStart = ({
  sourceX,
  sourceY,
  sourcePosition,
}: WrapEdgeProps): JSX.Element | null => {
  return (
    <circle
      className="react-flow__edgeupdater"
      cx={shiftX(sourcePosition, sourceX)}
      cy={shiftY(sourcePosition, sourceY)}
      r={radiusEdgeUpdater}
      stroke="transparent"
      fill="transparent"
    />
  );
}

export const EdgeAnchorEnd = ({
  targetX,
  targetY,
  targetPosition,
}: WrapEdgeProps): JSX.Element | null => {
  return (
    <circle
      className="react-flow__edgeupdater"
      cx={shiftX(targetPosition, targetX)}
      cy={shiftY(targetPosition, targetY)}
      r={radiusEdgeUpdater}
      stroke="transparent"
      fill="transparent"
    />
  );
}
```

Reply:
![Selection_015](https://user-images.githubusercontent.com/380645/109306932-90f6bd80-7848-11eb-8f39-91d8020f09a1.png)